### ISSUE TITLE
Bug in watch.js: Doesn't fire if 2 events are less than 1 second apart

### ISSUE THREAD

**Issue by mirismaili** [madewithkoji/koji-tools#15]
Hello,

The algorithm used in [watch.js](https://github.com/madewithkoji/koji-tools/blob/a1ca3c2ec52d5c8321afd128e44978494341df44/watch.js) to handle near events (less than 1 second apart) together, has a bug. Such way the second (and third, fourth, ...) event is (are) not handled at all!

https://github.com/madewithkoji/koji-tools/blob/a1ca3c2ec52d5c8321afd128e44978494341df44/watch.js#L26-L33

This is the first thing in the handler callback (and there is nothing before it):
```js
if (fsWait) return;
```
So the question is simple: What happens when `fsWait` is not falsy? The answer is simple too. The function just returns! And **the event**? Who will handle it? It goes to a [black hole](https://en.wikipedia.org/wiki/Black_hole).

**Comment by mirismaili** [comment]
The right practice is to some kind of *buffering*. This means you should handle any event immediately, by default. But this *default* can be broken in some special conditions. The special condition is the same as ***"to be less than 1 second after the last handled event"***.

In this condition, the events will be buffered (stored) in a memory and will be handled after expiring special condition, together.

**Comment by sthielen** [comment]
For this one + #16 and #17, can we instead just bring in https://github.com/paulmillr/chokidar as a dependency? I'd rather introduce something off-the-shelf than managing our own logic for something like this. Koji already uses chokidar for file-watching in the web editor's backend and it's been great. What do you think?

PS - Really appreciate these PRs and issues!!!

**Comment by mirismaili** [comment]
@sthielen 

I think that could be a good change. I've worked with *watchers* before in Java and I know its potential issues and problems. In my little experience with them in Javascript (in this project), I saw a similar issue that reminds me of them: *"Often reports events twice!"* (that being said [here](https://github.com/paulmillr/chokidar#why) too) 

I know dealing with these issues is not easy work. Especially because of the variety of file-systems across platforms. 

I couldn't find a suitable ready-to-use library for Java at that time and I'm glad to see a (seems to be) powerful library here on npm. 

PS - You're welcome. Hope to be useful.

**Comment by mirismaili** [comment]
@sthielen 

I was added and used [chokidar](https://github.com/paulmillr/chokidar) in a new branch. See #20.

2 notes:

1. The issues (#15, #16 and #17) are independent of using an external library or node's built-in way. So I branched out the new branch from this branch.

2. I was wrong about *"Often reports events twice"*! It wasn't about used *watching* system. It was from `git` command and can be resolved through a caching system. I was about implementing that as a performance-improvement and now should consider it as a bug-fix, as well.
